### ISSUE TITLE
Box loss: IoU and L1

### ISSUE THREAD

**Issue by Labaien96** [facebookresearch/detr#75]
I'm a little confused about how the bounding box loss is computed. In my case, I'm working with a custom dataset in which the targets are a dictionary containing bounding boxes, labels, and masks, just like in COCO. In my case, the bounding boxes are in the format [x_min,y_min,x_max,y_max] (I have put them in range 0-1 to train the model) and I don't know if I should change it to format [center_x, center_y, h, w] to compute the loss.  

``` 
def loss_boxes(self, outputs, targets, indices, num_boxes):
        """Compute the losses related to the bounding boxes, the L1 regression loss and the GIoU loss
           targets dicts must contain the key "boxes" containing a tensor of dim [nb_target_boxes, 4]
           The target boxes are expected in format (center_x, center_y, h, w), normalized by the image size.
        """
        assert 'pred_boxes' in outputs
        idx = self._get_src_permutation_idx(indices)
        src_boxes = outputs['pred_boxes'][idx]
        target_boxes = torch.cat([t['boxes'][i] for t, (_, i) in zip(targets, indices)], dim=0)

        loss_bbox = F.l1_loss(src_boxes, target_boxes, reduction='none')

        losses = {}
        losses['loss_bbox'] = loss_bbox.sum() / num_boxes

        loss_giou = 1 - torch.diag(box_ops.generalized_box_iou(
            box_ops.box_cxcywh_to_xyxy(src_boxes),
            box_ops.box_cxcywh_to_xyxy(target_boxes)))
        losses['loss_giou'] = loss_giou.sum() / num_boxes
        return losses
```

Looking at the code, you can see that the L1 loss is calculated without making any changes to the outputs or the targets, and on the other hand, the IoU is calculated by passing both of them from cxcywh format to xyxy format, so I deduce that the outputs of the model are in cxcywh format, the same as the targets. So, how should I pass the targets? In xyxy format or in cxcywh format? 

I have tried to pass it in xyxy format and then at the time of computing the loss, leave it the way it is to compute IoU, and change it to cxcywh format to compute the L1 loss but it doesn't go down much as the epochs go on.

**Comment by hafiz703** [comment]
As the documentation states "The target boxes are expected in format (center_x, center_y, h, w)" - isn't it supposed to be (center_x, center_y, **w**, **h**) as per box_xyxy_to_cxcywh function?

**Comment by fmassa** [comment]
@hafiz703 that is correct we expect `(center_x, center_y, w, h)`, could you send a PR fixing the documentation?
